### DISCUSSION TITLE
Problem routing between RoadWarriors virtual IPs

### DISCUSSION THREAD

**Discussion by Hellenya** [strongswan/strongswan#205] [Q&A]
Dear erveryone,

First of all i want say that I'm just a passionate user not a networking guru and i'm discovering the world of vpn... It has few days i'm woking on the following scenario which (i thought) would be quite cool for my needs. Now it's been 4 full days tests and investigations without success except if i can consider a headache as a success ;-)

My scenario is the following :

Standard RWs <==> (possible NAT) <==> Internet <==> VPS GW (with public static IP) <==> Internet <==> (possible NAT) <==> Specific RWs

- Standard RWs shoud get virtual IP from 172.16.1.0/24
- Specific RWs shoud get virual IP from 172.16.2.0/24
- All RWs will use split tunneling in order to keep their access to standard internet sites
- I would like that Standard RW can communicate with Specific RW
- I would like that Specific RW can communicate between themselves
- (optionnaly) I don't want the standard RW to be able to communicate between themselves

What seems to work :
- I've managed to create the connections between Standard RWs to the GW
- I've managed to create the connections between Specific RWs to the GW

My problem :
- Impossible to make any communication between RW (either Standard nor Specific)

My investigations brought me to see the following things :
- When i tcpdump (tcpdump -n -s0 -p -i eth0 esp or udp port 4500') on the server while i try to ping from a RW to another i can see traffic coming from the originator of ping but it seems that the GW doesn't know what to do with the incoming traffic.

I've read the wiki page about Forwarding and Split-Tunneling (and many other ones) but maybe some elements are out of my scope of understanding...

Here you will find files from the different machines for the test case in which i try to make Specific RW (zeus & pi) able to communicate between themselves on the subnet 172.16.2.0/24

_Note : for the moment zeus and pi are behind same NAT to access internet. But i've also tried with the same configuration to put a RW from outside this NAT without success_

Thanks by advance for your help, hope it's just a noob error that will not take too much time to analyze ;-)

Pierre-Jean

* vps = the GW
* zeus, pi = Specific RW

[pi.ipconf.statusall.txt](https://github.com/strongswan/strongswan/files/6015436/pi.ipconf.statusall.txt)
[pi.ipsec.conf.txt](https://github.com/strongswan/strongswan/files/6015437/pi.ipsec.conf.txt)
[pi.iptables-save.txt](https://github.com/strongswan/strongswan/files/6015438/pi.iptables-save.txt)
[vps.ipsec.conf.txt](https://github.com/strongswan/strongswan/files/6015439/vps.ipsec.conf.txt)
[vps.iptables-save.txt](https://github.com/strongswan/strongswan/files/6015440/vps.iptables-save.txt)
[vps.strongswan.statusall.txt](https://github.com/strongswan/strongswan/files/6015441/vps.strongswan.statusall.txt)
[vps.tcpdump.txt](https://github.com/strongswan/strongswan/files/6015442/vps.tcpdump.txt)
[zeus.ipsec.conf.txt](https://github.com/strongswan/strongswan/files/6015443/zeus.ipsec.conf.txt)
[zeus.iptables-save.txt](https://github.com/strongswan/strongswan/files/6015444/zeus.iptables-save.txt)
[zeus.strongswan.statusall.txt](https://github.com/strongswan/strongswan/files/6015445/zeus.strongswan.statusall.txt)

**Comment by tobiasbrunner** [comment]
Your NAT rules on the server are quire a mess. You might want to clean them up and make sure traffic from the roadwarriors is not affected (I guess it shouldn't even with the current setup but it really doesn't look good).

Your `tcpdump` output shows ESP traffic in both directions and to different UDP ports, so that does actually seem correct (ICMP request from one client to the other and a response back). Traffic counters in the status output are all 0, though. So something does not add up here. Make sure you generate some traffic before dumping the status output and check the numbers.

**Reply by Hellenya** [reply]
Thanks a lot for the help Tobias, i'm gonna check that @home as soon as i can.

For the NAT on the server It's possible that the different tries i made have a little messed up the setup,. I'll will do a fresh (re)install of th server to be sure to start from a clean setup and i'll update informations according to that.

Pierre-Jean

**Comment by Hellenya** [comment]
Hi,
I've done a full reinstallation of the vps (still not working anymore). The iptables-save on the server is as follow with 2 clients connected :
```
Generated by iptables-save v1.8.4 on Wed Feb 24 18:13:11 2021
*filter
:INPUT ACCEPT [1829:159113]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [1281:313300]
-A INPUT -s 82.121.250.172/32 -d 152.228.130.157/32 -i eth0 -p ipv4 -m policy --dir in --pol ipsec --reqid 4 --proto esp -j ACCEPT
-A INPUT -s 82.121.250.172/32 -d 152.228.130.157/32 -i eth0 -p ipv4 -m policy --dir in --pol ipsec --reqid 4 --proto esp -j ACCEPT
-A INPUT -s 82.121.250.172/32 -d 152.228.130.157/32 -i eth0 -p ipv4 -m policy --dir in --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A INPUT -s 82.121.250.172/32 -d 152.228.130.157/32 -i eth0 -p ipv4 -m policy --dir in --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A FORWARD -s 172.16.2.1/32 -d 172.16.2.0/24 -i eth0 -m policy --dir in --pol ipsec --reqid 4 --proto esp -j ACCEPT
-A FORWARD -s 172.16.2.0/24 -d 172.16.2.1/32 -o eth0 -m policy --dir out --pol ipsec --reqid 4 --proto esp -j ACCEPT
-A FORWARD -s 172.16.2.1/32 -d 172.16.1.0/24 -i eth0 -m policy --dir in --pol ipsec --reqid 4 --proto esp -j ACCEPT
-A FORWARD -s 172.16.1.0/24 -d 172.16.2.1/32 -o eth0 -m policy --dir out --pol ipsec --reqid 4 --proto esp -j ACCEPT
-A FORWARD -s 172.16.2.2/32 -d 172.16.2.0/24 -i eth0 -m policy --dir in --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A FORWARD -s 172.16.2.0/24 -d 172.16.2.2/32 -o eth0 -m policy --dir out --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A FORWARD -s 172.16.2.2/32 -d 172.16.1.0/24 -i eth0 -m policy --dir in --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A FORWARD -s 172.16.1.0/24 -d 172.16.2.2/32 -o eth0 -m policy --dir out --pol ipsec --reqid 3 --proto esp -j ACCEPT
COMMIT
Completed on Wed Feb 24 18:13:11 2021
Generated by iptables-save v1.8.4 on Wed Feb 24 18:13:11 2021
*security
:INPUT ACCEPT [1530:134481]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [1281:313300]
COMMIT
Completed on Wed Feb 24 18:13:11 2021
Generated by iptables-save v1.8.4 on Wed Feb 24 18:13:11 2021
*raw
:PREROUTING ACCEPT [1841:160361]
:OUTPUT ACCEPT [1281:313300]
COMMIT
Completed on Wed Feb 24 18:13:11 2021
Generated by iptables-save v1.8.4 on Wed Feb 24 18:13:11 2021
*mangle
:PREROUTING ACCEPT [1841:160361]
:INPUT ACCEPT [1841:160361]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [1281:313300]
:POSTROUTING ACCEPT [1281:313300]
COMMIT
Completed on Wed Feb 24 18:13:11 2021
Generated by iptables-save v1.8.4 on Wed Feb 24 18:13:11 2021
*nat
:PREROUTING ACCEPT [509:34872]
:INPUT ACCEPT [199:9052]
:POSTROUTING ACCEPT [96:6464]
:OUTPUT ACCEPT [96:6464]
COMMIT
Completed on Wed Feb 24 18:13:11 2021
```
Then i activated the log of firewalld and captured the kernel traces during ping trial between 172.16.2.2 to 172.16.2.1 :
```
Feb 24 18:15:01 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=26253 PROTO=4 
Feb 24 18:15:02 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=26847 PROTO=4 
Feb 24 18:15:03 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=27738 PROTO=4 
Feb 24 18:15:04 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=28023 PROTO=4 
Feb 24 18:15:05 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=28420 PROTO=4 
Feb 24 18:15:06 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=28997 PROTO=4 
Feb 24 18:15:07 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=29930 PROTO=4 
Feb 24 18:15:08 vps-bd9da615 kernel: FINAL_REJECT: IN=eth0 OUT= MAC=fa:16:3e:b7:16:b6:0a:74:5d:c7:7c:43:08:00 SRC=82.121.250.172 DST=152.228.130.157 LEN=104 TOS=0x00 PREC=0x00 TTL=45 ID=30447 PROTO=4
```
The ping traffic seems to be rejected but I can't figure why... If someone has any suggestion... ^^

Thanks again

**Reply by tobiasbrunner** [reply]
> Then i activated the log of firewalld and captured the kernel traces during ping trial between 172.16.2.2 to 172.16.2.1 :

Did that change anything else? These log messages may just be caused by a log rule before an _expected_ reject rule that you don't actually have.

What do the traffic stats in `ipsec statusall` say (on both hosts)?

Probably not directly related, but this looks weird: `-p ipv4`. What traffic selectors (`left|rightsubnet` and perhaps `left|rightprotoport`) did you configure now?
